File: java/P376.java
class Solution {
    public int wiggleMaxLength(int[] nums) {
        if (nums == null) return 0;
        if (nums.length <= 1) return nums.length;
        int increase = 1, decrease = 1;
        for (int i = 1; i < nums.length; i++) {
            if (nums[i] > nums[i-1]) increase = decrease+1;
            else if (nums[i] < nums[i-1]) decrease = increase+1;
        }
        return Math.max(increase, decrease);
    }
}

File: python/P376.py
class Solution(object):
    def wiggleMaxLength(self, nums):
        size = len(nums)
        inc = dec = 1
        for x in range(1, size):
            if nums[x] > nums[x-1]:
                inc = dec+1
            elif nums[x] < nums[x-1]:
                dec = inc+1
        return max(dec, inc) if size else 0 
        